packages/pwa-kit-react-sdk/src/ssr/universal/utils.js

```diff
@@ -51,3 +51,64 @@ export const getProxyConfigs = () => {
     // Clone to avoid accidental mutation of important configuration variables.
     return configs.map((config) => ({...config}))
 }
+
+/**
+ * Returns the express app configuration file in object form. The object resolution will
+ * be as follows (from highest to lowest priority):
+ *
+ * > {target_name}.ext
+ * > local.ext
+ * > default.ext
+ * > package.json (mobify.key)
+ *
+ * Each file marked with `ext` can optionally be terminated with `yml`, `yaml` or
+ * `json` in that priority.
+ *
+ * NOTE: This is an isomorphic function, when run on the browser the config returned will
+ * be the value serialized in the html.
+ *
+ * @returns - the application configuration object.
+ */
+/* istanbul ignore next */
+export const getConfig = () => {
```

Comment:
I'm a little bit concerned about the performance of this util function, specifically on the server side.

This util could potentially be called from user land many times during ssr, and each time this util gets called, it will dynamically require cosmiconfig and do a `cosmiconfigSync` file search. I'm not exactly sure the magnitude of the operations but is it possible to cache it, so that we technically only load configs **once** at server bootup.

Reply:
`cosmiconfig` has built in caching. You can read about it [here](https://www.npmjs.com/package/cosmiconfig#caching). I think it does what we need it to do.